workflow: Cluster UI autopublishing fix

nathanstilwell opened the pull request:
Using action `actions/setup-node@v3` we need to provide specific configuration for the internally generated `.npmrc`. This commit sets `registry-url` and `always-auth` to be explicit and changes the environment variable `NPM_TOKEN` to `NODE_AUTH_TOKEN` since this is what the action uses internally to set the auth token. I'm also switching back to using `npm` to publish as the setup-node action uses Yarn2 internally which can not read npm configuration and must have its own `.yarnrc.yml` generated.

A couple more improvements were made (since I was here),
- Add explicit output when a package should be published
- Check for the existence of a git tag before creating one (to prevent duplicates)
- Adds `--ignore-scripts` option to publish command to prevent duplicate builds. 

Test published for this PR, [publish_cluster_ui #17](https://github.com/cockroachdb/cockroach/actions/runs/4027808329/jobs/6923975372) published [22.2.1-publishtest.0](https://www.npmjs.com/package/@cockroachlabs/cluster-ui/v/22.2.1-publishtest.0) (along with git [tag](https://github.com/cockroachdb/cockroach/releases/tag/%40cockroachlabs%2Fcluster-ui%4022.2.1-publishtest.0)) 

Epic: none
Release note: None

Commit:
workflow: Cluster UI autopublishing fix

Using action `actions/setup-node@v3` we need to provide specific configuration
for the internally generated `.npmrc`. This commit sets `registry-url`
and `always-auth` to be explicit and changes the environment variable
`NPM_TOKEN` to `NODE_AUTH_TOKEN` since this is what the action uses
internally to set the auth token. I'm also switching back to using `npm`
to publish as the setup-node action uses Yarn2 internally which can not
read npm configuration and must have its own `.yarnrc.yml` generated.

A couple more improvements were made (since I was here),
- Add explicit output when a package should be published
- Check for the existence of a git tag before creating one (to prevent
  duplicates)
- Adds `--ignore-scripts` option to publish command to prevent duplicate
  builds.
Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/96120)
<!-- Reviewable:end -->

maryliag reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96120#-:-NMoxFEtE9QbhDEoyNLT:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sjbarag)



<!-- Sent from Reviewable.io -->

Commit:
ui: Added "files" to cluster-ui package.json

Adding a "files" property to the package.json of Cluster UI to ensure
that all the types are included in the publish.

Epic: none
Release note: None
